wire-ios/Wire-iOS/Sources/Helpers/SystemSaveFilePresenter.swift

```diff
@@ -0,0 +1,58 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import UIKit
+
+protocol SystemSaveFilePresenting {
+    func presentSystemPromptToSave(file fileURL: URL, completed: @escaping () -> Void)
+}
+
+final class SystemSavePresenter: NSObject, SystemSaveFilePresenting {
+    private var finishedPresenting: (() -> Void)?
+    private let documentInteractionController: UIDocumentInteractionController
+
+    init(documentInteractionController: UIDocumentInteractionController = UIDocumentInteractionController()) {
+        self.documentInteractionController = documentInteractionController
+        super.init()
+    }
+
+    func presentSystemPromptToSave(file fileURL: URL, completed: @escaping () -> Void) {
+        documentInteractionController.delegate = self
+        self.finishedPresenting = completed
+        documentInteractionController.url = fileURL
+        documentInteractionController.name = fileURL.lastPathComponent
+        documentInteractionController.presentPreview(animated: true)
+    }
+}
+
+extension SystemSavePresenter: UIDocumentInteractionControllerDelegate {
+
+    func documentInteractionControllerViewControllerForPreview(
+        _ controller: UIDocumentInteractionController,
+        finished: @escaping () -> Void
+    ) -> UIViewController {
+        self.finishedPresenting = finished
+        return UIApplication.shared.topmostViewController(onlyFullScreen: false)!
```

Comment:
Using `!` here is dangerous..

```suggestion
        guard let viewController = UIApplication.shared.topmostViewController(onlyFullScreen: false) else {
            assertionFailure("expected to access Application top view controller!")
            return UIViewController()
        }
        return viewController
```

Reply:
Yeah thought about that. However, this is the pattern we using across the app. Going with that for now. Moreover, returning `UIViewController` without any screen bounds doesn't provide us with any benefit here.

Reply:
`UIApplication.shared.topmostViewController(onlyFullScreen: false)` should not return an optional. May be we should have a greater discussion outside of the scope of this PR.

Reply:
The benefit is, that it doesn't crash the app. I personally don't like to approve code crashing the app, that can be avoided by a simple check and the argument that other places in the app have the same pattern doesn't make it valid here.

Reply:
done